### DISCUSSION TITLE
0.43.0 beta

### DISCUSSION THREAD

**Discussion by rejetto** [rejetto/hfs#199] [Announcements]
## what's new
- HFS will include translations, because I noticed that keeping language files updated separately is a burden. You will still be able to add and override translations with extra files.

- you are now able to force a specific language instead of leaving it to the browser to decide (Admin > Language)

- even if you have "menu on link" enabled, that will apply only to files. Folders will navigate by clicking, and offer a button for menu

- import csv for accounts now support "redirect" field

- an icon will mark files that require extra permission

- force https option

- many fixes

beta 4
[hfs-windows.zip](https://github.com/rejetto/hfs/files/11159658/hfs-windows.zip)
[hfs-mac-arm.zip](https://github.com/rejetto/hfs/files/11159657/hfs-mac-arm.zip)
[hfs-mac.zip](https://github.com/rejetto/hfs/files/11159655/hfs-mac.zip)
[hfs-linux.zip](https://github.com/rejetto/hfs/files/11159656/hfs-linux.zip)

**Comment by gogi100** [comment]
i noticed two problems:

when a user clicks on the icon next to the folder, the picture below
![icon](https://user-images.githubusercontent.com/21057856/230199473-6eb18422-3e77-4a1d-b109-e840aac743c1.jpg)

its open menu like on picture below
![foldermenu](https://user-images.githubusercontent.com/21057856/230199588-0dbdf073-83bf-46e3-ad27-64b4b2b5aeca.jpg)

the first problem, this menu is not translated on language that i'm choosed in admin-panel.
the second problem its shows download option, but i set option that user cannot download files and folders.

one more question where is button that i can choose that i disable creating/uploading folder/s?

**Comment by rejetto** [comment]
translation works, i guess it's the hfs-lang-sr.json file that is not updated.
I can take care only of english and italian languages.
This is italian version
<img width="160" alt="image" src="https://user-images.githubusercontent.com/1367199/230200826-cba955ae-c51c-466b-9a42-cd4a2938951d.png">

the download button is visible but of course doesn't work in your case.
it is displayed because currently it doesn't distinguish the case "no one can download"  with the case of "if you enter right credentials you can download" .

The upload question is already answered in discussion #194 .

**Comment by gogi100** [comment]
**translation works, i guess it's the hfs-lang-sr.json file that is not updated.**

what i must translate in my json file?

**Reply by rejetto** [reply]
you can compare the english text you see in the dialog with the content of this file
https://github.com/rejetto/hfs/blob/main/langs/hfs-lang-en.json

if you click "history" you will see changes of english in time.

**Comment by rejetto** [comment]
[beta6](https://github.com/rejetto/hfs/files/11163204/hfs-windows.zip)

@gogi100 in this release it will not show download button if NO ONE can download

**Reply by gogi100** [reply]
the download button still appears in beta 6
![downloadbutton](https://user-images.githubusercontent.com/21057856/230279848-d7795810-bd33-4a70-a3b6-fd754d89edc6.jpg)

**Reply by gogi100** [reply]
the permissions in admin-panel for the folder 'deljeni1' where is the folder '18' are like on picture
![permissions](https://user-images.githubusercontent.com/21057856/230281585-0bdd86f4-f5b3-4a48-aa11-51ca103f5a91.jpg)

the inheritance does not exist.

**Reply by gogi100** [reply]
the same, when I change 'who can download' to 'no one' the button 'download' is displayed

**Reply by rejetto** [reply]
yes, the thing for "download" is working only for files, while on folders it is considering only the "list" permission

**Reply by rejetto** [reply]
testing download for folders is much more complicated, but i'm trying

**Reply by rejetto** [reply]
ok, maybe i got it. Try now

[beta7](https://github.com/rejetto/hfs/files/11179903/hfs-windows.zip)

**Reply by gogi100** [reply]
perfect, one more thank you

**Comment by gogi100** [comment]
Thanks, for help
